### ISSUE TITLE
[AllowNull]/[DisallowNull] annotations on extension methods' parameters aren't respected in method groups

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#48143]
**Version Used**: 
```
Branch master (26 Sep 2020)
Latest commit 3e38d76 by CyrusNajmabadi:
Merge pull request #48002 from CyrusNajmabadi/implicitCreation

Use implicit object creation in the IDE layers.
```

**Steps to Reproduce**:

```cs
#nullable enable
using System;
using System.Diagnostics.CodeAnalysis;

class Test
{
  public static void M1(string? s)
  {
    Action a = s.AllowNullExtension; // false warning
  }

  public static void M2(string? s)
  {
    Action a = s.DisallowNullExtension; // missing warning
  }
}

public static class Extension
{
   public static void AllowNullExtension([AllowNull] this string p) { }
   public static void DisallowNullExtension([DisallowNull] this string? p) { }
}
```

**Expected Behavior**:
Warning for `Action a = s.DisallowNullExtension;`. No warnings for `Action a = s.AllowNullExtension;`

**Actual Behavior**:
The opposite. Roslyn incorrectly warns about nullable qualifier for extension method group that allows nullable `this` arguments and doesn't warn about the method group that explicitly disallows nullable arguments

**Comment by TessenR** [comment]
Sorry I think I either used incorrect version or minified the example too much, can't reproduce it right now anymore. Will reopen if I'll be able to reproduce it again
